nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -788,7 +788,7 @@ in
 
       boot.loader.grub.devices = mkIf (cfg.device != "") [ cfg.device ];
 
-      boot.loader.grub.mirroredBoots = mkIf (cfg.devices != [ ]) [
+      boot.loader.grub.mirroredBoots = mkIf (cfg.devices == [ ]) [
         {
           path = "/boot";
           inherit (cfg) devices;
```

Comment:
If the list of devices is empty, why do we even need to pass them here?

Reply:
Indeed that makes no sense. Like I said in the PR cover message I think this block can be removed completely since the condition was false for most setups and nobody seemed to have noticed this problem before me (as I use mirroredBoots with two separate devices and partitions and had to configure it manually).

Reply:
Actually. I think the code is correct, are you using disko by chance? Because disko is setting `disko.devices` by default. But maybe it should actually set `boot.loader.grub.mirroredBoots` instead.

Reply:
Yeah I am using disko and I had a look again and something else is going on here. I agree that the logic is correct and my PR is wrong.

Reply:
We can probably improve disko to support this usecase better, but for the time beeing `lib.mkForce` on your own definition will overwrite what disko does.